src/Avalonia.Base/Reactive/DisposeOnNextObservable.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using Avalonia.Threading;
+
+namespace Avalonia.Reactive
+{
+    public class DisposeOnNextObservable<T> : LightweightObservableBase<T>, IObserver<T> where T : IDisposable
+    {
+        private IDisposable lastValue;
+
+        private void ValueNext(T value)
+        {
+            this.PublishNext(value);
+            lastValue?.Dispose();
+            lastValue = value;
```

Comment:
I'm having problems making `WeakReference` work with this. I dont want to use this direct-store method as it looks hackish...